core/types/src/key_request.rs

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) 2022 The MobileCoin Foundation
+
+//! SGX key request rust types
+
+use crate::{new_type_wrapper, Attributes, MiscellaneousSelect};
+use mc_sgx_core_sys_types::{
+    sgx_config_svn_t, sgx_cpu_svn_t, sgx_isv_svn_t, sgx_key_id_t, sgx_key_request_t,
+};
+
+new_type_wrapper! {
+    CpuSvn, sgx_cpu_svn_t;
+}
+
+impl Default for CpuSvn {
+    fn default() -> Self {
+        CpuSvn(sgx_cpu_svn_t { svn: [0; 16] })
+    }
+}
+
+new_type_wrapper! {
+    KeyRequest, sgx_key_request_t;
+}
+
+impl Default for KeyRequest {
+    fn default() -> Self {
+        KeyRequest(sgx_key_request_t {
+            key_name: 0,
+            key_policy: 0,
+            isv_svn: 0,
+            reserved1: 0,
+            cpu_svn: CpuSvn::default().into(),
+            attribute_mask: Attributes::default().into(),
+            // TODO ideally this should be random
```

Comment:
Not fond of this.  This goes back to our key_id, but to do a _good_ default it seems like we need someway for people to specify an RNG

Reply:
If `Default` is a footgun, then we shouldn't implement it. Based on the setter comment, should we just make a new() method that takes the relevant details?

Like,
```rust
impl KeyRequest {
    pub fn new<R: CryptoRng + RngCore>(csprng: &mut R, key_name: KeyName, ...) -> Self {
        /* ... */
    }
}
```

(I assume that key_policy is supposed to be either an enum or bitflags, even if `key_name` isn't...)

Reply:
Add `new()` method which takes in a `csprng`.  The new type does not take other parameters, left those for the builder pattern.  I could see an argument for `key_name` and `key_policy` being part of the method.

Also updated `key_name` to be an enum and `key_policy` to support bitwise operations.